misc/python/materialize/checks/range.py

```diff
@@ -0,0 +1,171 @@
+
+            > CREATE SOURCE range_source
+              FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-ranges-${testdrive.seed}')
+              FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+              ENVELOPE NONE
+            """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        manipulation = """
+                > CREATE MATERIALIZED VIEW $view_name$ AS
+                  SELECT
+                    index,
+                    i4_range,
+                    i8_range,
+                    num_range,
+                    ts_range,
+                    tstz_range,
+                    d_range,
+                    '[2,5]'::INT8RANGE AS i8_range2,
+                    INT8RANGE(2,4) AS i8_range3,
+                    GREATEST(i8_range, '[2,5]'::INT8RANGE) AS i8_greatest,
+                    LEAST(i8_range, '[2,5]'::INT8RANGE) AS i8_least,
+                    NULLIF(i8_range, '[2,17)'::INT8RANGE) AS i8_nif,
+                    LOWER(i8_range) AS i8_low,
+                    UPPER(i8_range) AS i8_up,
+                    ISEMPTY(i8_range) AS i8_empty,
+                    LOWER_INC(i8_range) AS i8_lowinc,
+                    UPPER_INC(i8_range) AS i8_upinc,
+                    LOWER_INF(i8_range) AS i8_lowinf,
+                    UPPER_INF(i8_range) AS i8_upinf,
+                    i8_range < '[2,16]'::INT8RANGE AS i8_lt,
+                    i8_range <= '[2,16]'::INT8RANGE AS i8_le,
+                    i8_range = '[2,16]'::INT8RANGE AS i8_eq,
+                    i8_range >= '[2,16]'::INT8RANGE AS i8_ge,
+                    i8_range > '[2,16]'::INT8RANGE AS i8_gt,
+                    i8_range @> INT8RANGE(2,3) AS i8_containsrange,
+                    i8_range @> 16::INT8 as i8_containselem,
+                    i8_range <@ INT8RANGE(2,3) AS i8_inrange,
+                    16::INT8 <@ i8_range as i8_inelem,
+                    i8_range && INT8RANGE(4,12) AS i8_overlap,
+                    i8_range << INT8RANGE(400, 500) AS i8_leftof,
+                    i8_range >> INT8RANGE(0, 1) AS i8_rightof,
+                    i8_range &< INT8RANGE(0, 1) AS i8_notextright,
+                    i8_range &> INT8RANGE(0, 1) AS i8_notextleft,
+                    i8_range -|- INT8RANGE(100, 150) AS i8_adjacent,
+                    i8_range + '[3,20]'::INT8RANGE AS i8_merge,
+                    i8_range * '[8,20]'::INT8RANGE AS i8_intersec,
+                    i8_range - '[8,120]'::INT8RANGE AS i8_diff,
+                    (SELECT partition FROM range_source_progress LIMIT 1) AS progress_range
+                  FROM range_table;
+
+                > INSERT INTO range_table SELECT * FROM range_table WHERE index = 1;
+            """
+
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                manipulation.replace("$view_name$", "range_view1"),
```

Comment:
This is all good and in fact quite readable, so please do leave it as it is, but for the record the same can be achieved with format strings:
```
return [ Testdrive(f" range_view{i} ") for i in [1,2]]
```
or something along those ines.